code/game/machinery/bees_apiary.dm

```diff
@@ -1,97 +1,172 @@
 //http://www.youtube.com/watch?v=-1GadTfGFvU
-//i could have done these as just an ordinary plant, but fuck it - there would have been too much snowflake code
 
-#define HONEYCOMB_COST 15
+/*
+
+> apiary tray
+> angry-bee hive
+
+*/
 
 /obj/machinery/apiary
 	name = "apiary tray"
 	icon = 'icons/obj/hydroponics.dmi'
 	icon_state = "hydrotray3"
 	density = 1
 	anchored = 1
-	var/nutrilevel = 0
+	var/beezeez = 0//beezeez removes 1 toxic and adds 1 nutrilevel per cycle
+	var/nutrilevel = 0//consumed every round based on how many bees the apiary is sustaining.
 	var/yieldmod = 1
-	var/mut = 1
+	var/damage = 1
 	var/toxic = 0
-	var/dead = 0
-	var/health = -1
-	var/maxhealth = 100
+
 	var/lastcycle = 0
 	var/cycledelay = 100
-	var/beezeez = 0
-	var/swarming = 0
-	var/honey_level = 0
+	var/list/pollen = list()//mostly unused for now. but might come handy later if we want to check what plants where pollinated by an apiary's bees
+
+	var/queen_bees_inside = 0
+	var/worker_bees_inside = 0
+	var/list/bees_outside_hive = list()
 
-	var/bees_in_hive = 0
-	var/list/owned_bee_swarms = list()
 	var/hydrotray_type = /obj/machinery/portable_atmospherics/hydroponics
 
-	machine_flags = FIXED2WORK | WRENCHMOVE
+	var/obj/item/weapon/reagent_containers/glass/consume = null
+
+	var/wild = 0
+
+	machine_flags = WRENCHMOVE
 
-//overwrite this after it's created if the apiary needs a custom machinery sprite
 /obj/machinery/apiary/New()
 	..()
-	overlays += image('icons/obj/apiary_bees_etc.dmi', icon_state="apiary0")
+	overlays += image('icons/obj/apiary_bees_etc.dmi', icon_state="apiary")
+	create_reagents(100)
```

Comment:
It's not losing reagents, it's an investment. If you always harvest an apiary as soon as it reaches full capacity you'll never get any extra queens or jelly. Think of it as an investment.

And believe me, apiaries do take a while to fill up. You have time to run that errand to cargo.

Did you know that in the current Beekeeping 2.0 bees can produce honey without flowers? Just place an apiary, add beezeez, and honey will start dripping out a minute later.